third_party/move/move-model/src/builder/module_builder.rs

```diff
@@ -1281,6 +1281,7 @@ impl<'env, 'translator> ModuleBuilder<'env, 'translator> {
                 let mut result = et.translate_seq(&loc, seq, &result_type);
                 et.finalize_types();
                 result = et.post_process_placeholders(result.into_exp()).into();
+                result = et.rewrite_block_pattern(result.into_exp()).into();
```

Comment:
Can we (a) rename `post_process_placeholders` into `post_process_body` (b) fold the logic of `rewrite_block_pattern` into `post_process_body`. This way we avoid to rewrite the AST twice. Also easier to avoid to forget calling this function.

Reply:
done